src/account/default.ts

```diff
@@ -1,61 +1,49 @@
-import assert from 'minimalistic-assert';
-
 import { ZERO } from '../constants';
-import { Provider, ProviderInterface } from '../provider';
+import { ProviderOptions } from '../provider';
+import { Provider } from '../provider/default';
 import { Signer, SignerInterface } from '../signer';
 import {
   Abi,
-  AddTransactionResponse,
   Call,
   InvocationsDetails,
   InvocationsSignerDetails,
-  InvokeFunctionTransaction,
+  InvokeFunctionResponse,
   KeyPair,
   Signature,
-  Transaction,
 } from '../types';
 import { EstimateFee, EstimateFeeDetails } from '../types/account';
-import { sign } from '../utils/ellipticCurve';
-import {
-  computeHashOnElements,
-  feeTransactionVersion,
-  getSelectorFromName,
-  transactionVersion,
-} from '../utils/hash';
-import { BigNumberish, bigNumberishArrayToDecimalStringArray, toBN, toHex } from '../utils/number';
-import { encodeShortString } from '../utils/shortString';
+import { feeTransactionVersion, transactionVersion } from '../utils/hash';
+import { BigNumberish, toBN, toHex } from '../utils/number';
 import { compileCalldata, estimatedFeeToMaxFee } from '../utils/stark';
 import { fromCallsToExecuteCalldataWithNonce } from '../utils/transaction';
 import { TypedData, getMessageHash } from '../utils/typedData';
 import { AccountInterface } from './interface';
 
 export class Account extends Provider implements AccountInterface {
-  public address: string;
-
   public signer: SignerInterface;
 
   constructor(
-    provider: ProviderInterface,
-    address: string,
+    providerOrOptions: ProviderOptions | Provider,
+    public address: string,
     keyPairOrSigner: KeyPair | SignerInterface
   ) {
-    super(provider);
+    super(providerOrOptions);
     this.signer =
       'getPubKey' in keyPairOrSigner ? keyPairOrSigner : new Signer(keyPairOrSigner as KeyPair);
-    this.address = address;
   }
 
   public async getNonce(): Promise<string> {
     const { result } = await this.callContract({
       contractAddress: this.address,
       entrypoint: 'get_nonce',
+      calldata: [],
```

Comment:
should be the default in provider.callContract()

Reply:
Removed here https://github.com/0xs34n/starknet.js/pull/243/commits/34db683267b639adbd37eb097573eb7210cb4a58